lib/SIL/IR/SILLocation.cpp

```diff
@@ -135,144 +147,98 @@ DeclContext *SILLocation::getAsDeclContext() const {
   return nullptr;
 }
 
-SILLocation::DebugLoc SILLocation::decode(SourceLoc Loc,
-                                          const SourceManager &SM) {
-  DebugLoc DL;
+SILLocation::FilePosition SILLocation::decode(SourceLoc Loc,
+                                              const SourceManager &SM) {
+  FilePosition DL;
   if (Loc.isValid()) {
     DL.Filename = SM.getDisplayNameForLoc(Loc);
     std::tie(DL.Line, DL.Column) = SM.getPresumedLineAndColumnForLoc(Loc);
   }
   return DL;
 }
 
-void SILLocation::dump(const SourceManager &SM) const {
+SILLocation::FilePosition *SILLocation::getCompilerGeneratedLoc() {
+  static FilePosition compilerGenerated({0, 0, "<compiler-generated>"});
+  return &compilerGenerated;
```

Comment:
Ah, I was wondering how this would be implemented with FilePosition being allocated separately. I guess that works.